src/pybind/mgr/rook/module.py

```diff
@@ -373,24 +373,32 @@ def _service_add_decorate(self, typename, spec, func):
         return RookWriteCompletion(lambda: func(spec), None,
                     "Creating {0} services for {1}".format(typename, spec.name))
 
-    def add_stateless_service(self, service_type, spec):
-        # assert isinstance(spec, orchestrator.StatelessServiceSpec)
-        if service_type == "mds":
-            return self._service_add_decorate("Filesystem", spec,
-                                         self.rook_cluster.add_filesystem)
-        elif service_type == "rgw" :
-            return self._service_add_decorate("RGW", spec,
-                                         self.rook_cluster.add_objectstore)
-        elif service_type == "nfs" :
-            return self._service_add_decorate("NFS", spec,
-                                         self.rook_cluster.add_nfsgw)
-        else:
-            raise NotImplementedError(service_type)
+    def add_mds(self, spec):
+        return self._service_add_decorate("Filesystem", spec,
+                                          self.rook_cluster.add_filesystem)
+
+    def add_rgw(self, spec):
+        return self._service_add_decorate("RGW", spec,
+                                          self.rook_cluster.add_objectstore)
+
+    def add_nfs(self, spec):
+        return self._service_add_decorate("NFS", spec,
```

Comment:
I feel that replacing 2 small-sized method with 6 oneliner methods is a bit of an overkill. It smells like these methods are just proxies/wrappers to other methods/lambdas, right? I see a lot of boilerplate code here.

Reply:
I don't want to make big refactorings in the rook module just now.